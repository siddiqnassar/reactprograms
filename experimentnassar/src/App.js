import React from 'react';
import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
    <div>Hello from nassar</div>
    
    </div>
  );
}

export default App;
