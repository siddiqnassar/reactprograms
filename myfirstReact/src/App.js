import React from 'react';
import logo from './logo.svg';
import './App.css';

export const add=(x,y)=>{
  return x+y;
}
export const Button = ({ name, handleClick }) => {
  return (
    <input className="button" type="button" value={name} onClick={handleClick} />
  );
}

function mockFn(){
  return (
    <div>Hii</div>
  )
}
function App() {
  return (
    <div className="App">
      <h1>Hello World</h1>
      <button className="btn" onClick={()=>this.mockFn()}>Click here</button>
    </div>
  );
}
export default App;
