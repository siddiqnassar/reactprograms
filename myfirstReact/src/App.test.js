import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {add} from './App';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
const app =shallow(<App/>);

test('test add',()=>{
  const value=add(1,2);
  expect(value).toBe(3);
})
it('renders correctly', ()=> {
expect(app).toMatchSnapshot();

});
describe('Button', () => {
  it('should be defined', () => {
    expect(App).toBeDefined();
  });
 
 });
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
