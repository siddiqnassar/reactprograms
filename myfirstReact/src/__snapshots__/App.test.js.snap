// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;
